driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java

```diff
@@ -318,8 +318,8 @@ private void sendCommandMessage(final CommandMessage message, final CommandEvent
 
         try {
             message.encode(bsonOutput, sessionContext);
-            CommandEventSender commandEventSender = new CommandEventSender(message);
-            commandEventSender.sendStartedEvent(bsonOutput);
+            CommandEventSender commandEventSender = new CommandEventSender(message, bsonOutput);
+            commandEventSender.sendStartedEvent();
 
             if (sendCompressor == null || SECURITY_SENSITIVE_COMMANDS.contains(commandEventSender.getCommandName())) {
```

Comment:
Seems like a code smell to be using the `commandEventSender` to get the name for the compressor here.  Could the `CommandDocumentHandler` moved up to here and passed in to the event sender?

Reply:
You're absolutely right.  The new patch hopefully removes the stench.